net-p2p/retroshare/retroshare-0.6.7.ebuild

```diff
@@ -9,6 +9,9 @@ DESCRIPTION="F2F, private and secure file sharing application"
 HOMEPAGE="https://retroshare.cc"
 SRC_URI="http://download.opensuse.org/repositories/network:/retroshare/Debian_Testing/retroshare-common_${PV}.orig.tar.gz -> ${P}.tar.gz"
 S="${WORKDIR}/RetroShare"
+# REVIEW(setan): we could build from the link belowe to be closer to upstream
+# but this requires git modules
+# https://github.com/RetroShare/RetroShare/archive/refs/tags/{PV}.tar.gz
```

Comment:
Upstream uses the sources we are effectively using for the build to distribute the packages, the one created automatically by github is of no use, so please remove this comment or modify it according to the current situation.

Disclaimer upstream developer here.

Reply:
Let me get this straight:
The github source code distribution is not meant for building?

Reply:
Exactly, the archive you find on github, it's just something github generate automatically when we push a tag on the repository

Reply:
while the other one (the correct one which the ebuild uses) is properly generated with a script that is distributed along with the repository that collects all the submodules and pack them together so retroshare can be compiled

Reply:
I see.
I was mislead by the 
"""
As always we honour our Free as in Freedom software for a Free Society stance, so you can download the source code from the following links:
"""

Where previously I saw this as "If you want to build from source, it's fine: here you go" now I see "yep, we comply with the GPL".
Is that right?

I thought that we could move to the git sources because it has higher point version (0.6.7.2) and I find that enticing.

I accept the situation as it is but do not understand the reasoning behind the souce distribution in that way since it's basically github.

Reply:
The RetroShare code comprise many sub-modules each of then on its own repository, all of them free as in freedom. So if you want to fiddle with the source code, test latest versions etc. you can directly checkout the repository, checkout the submodules and fiddle with it, edit, compile etc, but it requires a bit of fiddling with git. When we do an "official" release we prepare a source tarball comprising all the submodule so distributions, packagers and people who want to compile by themselves can do it without any problem even if the internet is broken.
Aside from that github every-time we create a git tag automatically create a tarball in its own way that doesn't include the submodules, so it is not meant for packagers, if the website points to that tarball we should fix it ASAP (if you want you can make a PR on the website which happen to be hosted on github too ;) ).

Reply:
I could try but I am not sure where to point to. Debian testing of OBS?

Reply:
Use the same link we use for the ebuild ;)

Reply:
Also please let me know what you think about the `NOTE`

Reply:
done